### PR TITLE
Remove all autoload generation in autogen

### PR DESCRIPTION
**Pull request by aadi-stripe**
Removes autoload generation from Sorbet entirely as we don't need it anymore (Stripe has completely transitioned to path-based autoloading with Zeitwerk).

### Motivation

Cleanup unnecessary code

### Test plan

See included automated tests. Tested on Stripe codebase.

### CODE REVIEW

File: main/options/options.cc [hunk 1]
```diff
@@ -399,33 +397,6 @@ buildOptions(const vector<pipeline::semantic_extension::SemanticExtensionProvide
...
-                                    "Name of Ruby object on which root autoloads should be installed",
-                                    cxxopts::value<string>()->default_value("Object"));
-    options.add_options("advanced")("autogen-autoloader-samefile",
-                                    "Modules that should never be collapsed into their parent. This helps break cycles "
-                                    "in certain cases. (e.g. Foo::Bar::Baz)",
-                                    cxxopts::value<vector<string>>());
-    options.add_options("advanced")("autogen-autoloader-strip-prefix",
-                                    "Prefixes to strip from file output paths. "
-                                    "If path does not start with prefix, nothing is stripped",
-                                    cxxopts::value<vector<string>>());
```
[the remark this answers is not in the document]
Comment (jez) [reply main/options/options.cc:428]: Have we already stopped passing these flags in Stripe's codebase?

If not, can we make that change and land it before merging this change?

File: main/realmain.cc [hunk 2]
```diff
@@ -804,25 +730,21 @@ int realmain(int argc, char *argv[]) {
             gs->suppressErrorClass(core::errors::Resolver::StubConstant.code);
             gs->suppressErrorClass(core::errors::Resolver::RecursiveTypeAlias.code);
 
-            indexed = pipeline::package(*gs, move(indexed), opts, *workers);
```
Comment (jez) [inline main/realmain.cc:807]: Why are we deleting this?

File: packager/packager.cc [hunk 3]
```diff
@@ -1548,13 +1548,6 @@ vector<ast::ParsedFile> Packager::run(core::GlobalState &gs, WorkerPool &workers
 
     files = findPackages(gs, workers, std::move(files));
     setPackageNameOnFiles(gs, files);
-    if (gs.runningUnderAutogen) {
-        // Autogen only requires package metadata. Remove the package files.
-        auto it = std::remove_if(files.begin(), files.end(),
-                                 [&gs](auto &file) -> bool { return file.file.data(gs).isPackage(); });
-        files.erase(it, files.end());
-        return files;
-    }
```
Comment (jez) [inline packager/packager.cc:1557]: Should we `ENFORCE(!gs.runningUnderAutogen, "...")` with an explanation why?

**Review by jez — APPROVED** [review]

File: main/options/options.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (nroman-stripe) [reply main/options/options.cc:428]: ++ ^^^

File: packager/packager.cc (hunk 3 above)
Comment (aadi-stripe) [reply packager/packager.cc:1557]: Sure

File: main/options/options.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (aadi-stripe) [reply main/options/options.cc:428]: Yes, that will go first.

**Comment by aadi-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_NxLEUaExV0vd5A
→ https://go/builds/bui_NxLER1j5txzbIs
→ https://go/builds/bui_NxLE8lfFfqVJMF

File: main/realmain.cc [hunk 4]
```diff
@@ -804,25 +730,27 @@ int realmain(int argc, char *argv[]) {
...
+            // Remove all package files
+            auto it = std::remove_if(indexed.begin(), indexed.end(),
+                                     [&gs](auto &file) -> bool { return file.file.data(*gs).isPackage(); });
+            indexed.erase(it, indexed.end());
```
Comment (jez) [inline main/realmain.cc:736]: from slack:

"should we have solved this with `--ignore=__package.rb`?"

"ah true, we can do that"

**Pull request merged**